packages/dbml-core/src/parse/dbml/src/lib/result.ts

```diff
@@ -0,0 +1,34 @@
+import { ParsingError } from './errors';
+
+export default class Result<T> {
```

Comment:
It's better to add some note to explain what are this `Result` class used for? Is it the result of the parsing process or it could be used for other generic use cases?

Reply:
If it could be used for other generic use cases, does it make sense to hardcode the error type for this?